src/components/DataSourceFormBody/index.tsx

```diff
@@ -10,87 +13,109 @@ import DataSourceSetup from "@/components/DataSourceSetup";
 import DataModelGeneration from "@/components/DataModelGeneration";
 import ApiSetup from "@/components/ApiSetup";
 import { dataSourceForms, dbTiles } from "@/mocks/dataSources";
+import DataSourceStore from "@/stores/DataSourceStore";
+import type { DataSourceData } from "@/stores/DataSourceStore";
 
 import type { FC } from "react";
 
 interface DataSourceFormBodyProps {
-  step: number;
-  setStep: React.Dispatch<React.SetStateAction<number>>;
-  onFinish: (data: DataSourceForm) => void;
-  setState: React.Dispatch<React.SetStateAction<DataSourceForm>>;
-  formState?: DataSourceForm;
+  defaultData?:
+    | {
+        step: number;
+        apiSetup: DataSourceForm | null;
```

Comment:
`apiSetup` is ambiguous naming


```
formState: {
  step1: {
     data: ...
     initialValue: ....
  },
  step2: {
     data: ...
     initialValue: ....
  },
  step3: {
     data: ...
     initialValue: ....
  },
}
```

better to describe what is inside the form via props